src/main/java/com/purbon/kafka/topology/BuilderCLI.java

```diff
@@ -36,6 +40,12 @@
   public static final String VERSION_OPTION = "version";
   public static final String VERSION_DESC = "Prints useful version information.";
 
+  public static final String EXTRACT_OPTION = "extract";
+  public static final String EXTRACT_DESC = "Extract to current cluster state and outputs as JSON";
```

Comment:
why JSON, is an option to add a serialiser to use yaml as the original descriptor files? or even better introduce a flag to have both?

Reply:
Option to have both is fine as well. I tried to use separate just to prevent confusion between topology and 'flat descriptions' (which should rather be called clusterstate).